src/functional_data.Rmd

```diff
@@ -0,0 +1,177 @@
+## In this case we use column indices
+tsk1 = makeFDARegrTask(data = df, target = "V1", 
+  fd.features = list("UVVIS" = 3:136, "NIR" = 137:367),
+  fd.grids = list("UVVIS" = fuelSubset$uvvis.lambda,
+    "NIR" = fuelSubset$nir.lambda))
+
+## Create a regression task, classification tasks behave analogously
+## In this case we use the column names and assume that the data was 
+## measured at constant time intervals, thus we do not require 
+## setting fd.grids.
+tsk2 = makeFDARegrTask(data = df, target = "V1", 
+  fd.features = list("UVVIS" = stri_paste("V", 3:136),
+    "NIR" = stri_paste("V", 137:367)))
+```
+
+
+
+
+## Constructing a learner
+
+For functional data, [learners](&Learner.md) are constructed using 
+`makeLearner("<fdaclassif.<R_method_name>")` or 
+`makeLearner("<fdaregr.<R_method_name>")` depending on the target variable.
+
+Applying learners to a [FDATask](&makeFDATask) works in two ways:
+
+* Use a [learner](&Learner.md)
+that supports functional features, those learners have prefixes  __fdaclassif.__ or __fdaregr.__
+
+  + For regression: 
+```{r}
+## The following learners can be used for the task.
+## head(listLearners(tsk))
+## Create a FDboost learner
+fdalrn = makeLearner("fdaregr.FDboost")
+```
+
+  + And for classification:
+  
+```{r}
+## knn learner
+knn.lrn = makeLearner("fdaclassif.knn")
+```
+
+* Use a _standard_ [learner](&Learner.md): 
+In this case the temporal structure is disregarded
+
+* Alternatively, transform the functional data into a non-temporal/non-functional space by [extracting](functional_data.Rmd#feature-extraction) features before training.
+In this case, a normal regression- or classification- [learner](&Learner.md) 
+can be applied. 
+This is explained in more detail in the [feature extraction](functional_data.Rmd#feature-extraction)
+section below.
+
+## Train the learner
+
+The resulting learner can now be trained on the task created in section [Creating a FDATask](functional_data.Rmd#creating-a-fdatask) above. 
+
+```{r}
+# Train the fdalrn on the FDATask
+m = train(learner = fdalrn, task = tsk)
+p = predict(m, tsk)
+performance(p, rmse)
+
+# Or simply resample (3-fold CV) instead
+resample(fdalrn, tsk, resampling = cv3, measures =  mse)
+```
+
+Alternatively, learners that do not specifically treat functional covariates can
+be applied. In this case the temporal structure is completely disregarded, and all
+columns are treated as independent.
+
+```{r}
+# Train a normal learner on the FDATask
+rpart.lrn = makeLearner("regr.rpart")
+m = train(learner = lrn, task = tsk)
+```
+
+### Feature Extraction
+
+In contrast to applying a learner that works on a [FDATask](&makeFDATask), the [FDATask](&makeFDATask) can be converted to a normal [&Task.md].
+This works by transforming the functional features into a 
+non-functional domain, e.g by extracting wavelets.
+
+The currently supported preprocessing functions are:
+* discrete wavelet transform
```

Comment:
note: unrelated to tutorial
It would be really nice if these preprocessing methods could be used in forecasting as well. Maybe we could construct a sub preprocessing method like `createWaveletFeatures()` that can be used on `TimeTasks`?